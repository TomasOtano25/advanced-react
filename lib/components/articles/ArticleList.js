import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles, store }) => (
  <div>
    {Object.values(articles).map(article => (
      <Article key={article.id} article={article} store={store} />
    ))}
  </div>
);

ArticleList.propTypes = {
  articles: PropTypes.object.isRequired,
  store: PropTypes.object.isRequired,
};

export default ArticleList;
