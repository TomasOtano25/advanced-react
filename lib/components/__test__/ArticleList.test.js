import React from 'react';
import renderer from 'react-test-renderer';

import ArticleList from '../articles/ArticleList';

describe('ArticleList', () => {
  const testData = {
    articles: { a: { id: 'a' }, b: { id: 'b' } },
    articleActions: { lookupAuthor: jest.fn(() => ({})) },
  };

  it('renders conrrectly', () => {
    // const element = renderer.create(<div>Hello</div>).toJSON();

    const tree = renderer.create(<ArticleList {...testData} />).toJSON();

    // console.log(tree);
    expect(tree).toMatchSnapshot();
    expect(tree.children).toHaveLength(2);
    // u = actualizar instancia
  });
});
