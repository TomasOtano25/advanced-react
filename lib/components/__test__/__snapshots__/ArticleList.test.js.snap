// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders conrrectly 1`] = `
<div>
  <div
    className="sc-bdVaJa bcezof"
  >
    <h2
      className="sc-bwzfXH hZEuEv"
    />
    <div
      className="sc-htpNat csdQsP"
    />
    <a
      className="sc-bxivhb hMFYqw"
      href={undefined}
    >
      undefined undefined
    </a>
    <div
      className="sc-ifAKCX kjQGGW"
    />
    <br />
  </div>
  <div
    className="sc-bdVaJa bcezof"
  >
    <h2
      className="sc-bwzfXH hZEuEv"
    />
    <div
      className="sc-htpNat csdQsP"
    />
    <a
      className="sc-bxivhb hMFYqw"
      href={undefined}
    >
      undefined undefined
    </a>
    <div
      className="sc-ifAKCX kjQGGW"
    />
    <br />
  </div>
</div>
`;
